src/compass/utils/geo_metadata.py

```diff
@@ -265,3 +265,181 @@ def to_file(self, dst, fmt:str):
+                              {'description':'direction of orbit (ascending, descending)'})
+
+        # VRT params
+        add_dataset_and_attrs(metadata_group, 'tiff_path',
+                              np.string_(self.tiff_path),
+                              {'description': 'path to TIFF containing the burst'})
+        add_dataset_and_attrs(metadata_group, 'i_burst', self.i_burst,
+                              {'description': 'Sentinel platform ID (S1, S2)'})
+        # window parameters
+        add_dataset_and_attrs(metadata_group, 'range_window_type',
+                              np.string_(self.range_window_type),
+                              {'description': 'name of the weighting window type used during processing'})
+        add_dataset_and_attrs(metadata_group, 'range_window_coefficient',
+                              self.range_window_coefficient,
+                              {'description': 'value of the weighting window coefficient used during processing'})
+
+        geogrid_group = metadata_group.require_group('geogrid')
+        add_dataset_and_attrs(geogrid_group, 'start_x', self.geogrid.start_x,
+                              {'description': 'starting y coordinate of geogrid output',
+                               'units': 'consistent with EPSG'})
+        add_dataset_and_attrs(geogrid_group, 'start_y', self.geogrid.start_y,
+                              {'description': 'starting y coordinate of geogrid output',
+                               'units': 'consistent with EPSG'})
+        add_dataset_and_attrs(geogrid_group, 'spacing_x',
+                              self.geogrid.spacing_x,
+                              {'description': 'x spacing coordinate of geogrid output',
+                               'units': 'consistent with EPSG'})
+        add_dataset_and_attrs(geogrid_group, 'spacing_y',
+                              self.geogrid.spacing_y,
+                              {'description': 'y spacing coordinate of geogrid output',
+                               'units': 'consistent with EPSG'})
+        add_dataset_and_attrs(geogrid_group, 'width', self.geogrid.width,
+                              {'description': 'width of geogrid output',
+                               'units': 'consistent with EPSG'})
+        add_dataset_and_attrs(geogrid_group, 'length', self.geogrid.length,
+                              {'description': 'length of geogrid output',
+                               'units': 'consistent with EPSG'})
+        add_dataset_and_attrs(geogrid_group, 'epsg',
+                              self.geogrid.epsg,
+                              {'description': 'EPSG of output geogrid'})
+
+        add_dataset_and_attrs(metadata_group, 'nodata',
+                              np.string_(self.nodata),
+                              {'description': 'value used when no data present'})
+        add_dataset_and_attrs(metadata_group, 'input_data_ipf_version',
```

Comment:
I think that the version of the software used for processing should go in a separate metadata section. What do we do for NISAR?

Reply:
In `root` group? Or with `complex_backscatter` group?

I don't think we save it in NISAR; I was unable to find anything through `grep`.

Reply:
For NISAR, we do have a section under e.g., /science/LSAR/GUNW/metadata/processingInformation/algorithms where we have a metadata entry called ISCEversion. I guess what I mean is that for the CSLC metadata we should have a separate group (may be call it algorithms) where we list software version and corrections applied. For example, we should have an entry "antenna_pattern_correction" that tells the user if the antenna pattern correction has been applied. Does this make sense?